aten/src/ATen/cpu/vec/vec256/vec256_float.h

```diff
@@ -74,18 +74,26 @@ template <> class Vectorized<float> {
     return b;
   }
   static Vectorized<float> loadu(const void* ptr, int64_t count = size()) {
+    // A constant array to initialize an AVX2 register to be used as a 32-bit
+    // granularity mask.
+    alignas(64) constexpr int32_t masks[9][8] = {
+      // clang-format off
+      {  0,  0,  0,  0,  0,  0,  0,  0,  },
+      { -1,  0,  0,  0,  0,  0,  0,  0,  },
+      { -1, -1,  0,  0,  0,  0,  0,  0,  },
+      { -1, -1, -1,  0,  0,  0,  0,  0,  },
+      { -1, -1, -1, -1,  0,  0,  0,  0,  },
+      { -1, -1, -1, -1, -1,  0,  0,  0,  },
+      { -1, -1, -1, -1, -1, -1,  0,  0,  },
+      { -1, -1, -1, -1, -1, -1, -1,  0,  },
+      { -1, -1, -1, -1, -1, -1, -1, -1,  },
+      // clang-format on
+    };
     if (count == size())
       return _mm256_loadu_ps(reinterpret_cast<const float*>(ptr));
-    __at_align32__ float tmp_values[size()];
-    // Ensure uninitialized memory does not change the output value See https://github.com/pytorch/pytorch/issues/32502
-    // for more details. We do not initialize arrays to zero using "={0}" because gcc would compile it to two
-    // instructions while a loop would be compiled to one instruction.
-    for (auto i = 0; i < size(); ++i) {
-      tmp_values[i] = 0.0;
-    }
-    std::memcpy(
-        tmp_values, reinterpret_cast<const float*>(ptr), count * sizeof(float));
-    return _mm256_loadu_ps(tmp_values);
+    __m256i mask =
+        _mm256_load_si256(reinterpret_cast<const __m256i*>(masks[count]));
+    return _mm256_maskload_ps(reinterpret_cast<const float*>(ptr), mask);
   }
   void store(void* ptr, int64_t count = size()) const {
```

Comment:
Do you plan to use the same strategy for `store`?

Reply:
Good call, I should go ahead and do `store` while I'm here.